index.js

```diff
@@ -1,16 +1,37 @@
 'use strict';
 
+class PriorityQueue {
+	constructor() {
+		this._queue = [];
+	}
+	_compare(a, b) {
+		return b.priority - a.priority;
+	}
+	push(run, priority) {
+		priority = priority || 0;
+		this._queue.push({priority, run});
+		this._queue.sort(this._compare);
+	}
+	shift() {
+		return this._queue.shift().run;
+	}
+	get length() {
+		return this._queue.length;
+	}
+}
+
 class PQueue {
 	constructor(opts) {
 		opts = Object.assign({
-			concurrency: Infinity
+			concurrency: Infinity,
+			Queue: PriorityQueue
```

Comment:
Option name should be lowercase. Just `// eslint-disable-line rule-name` the warning on line 34.